sdk/core/azure-core/src/main/java/com/azure/core/util/HttpClientOptions.java

```diff
@@ -355,14 +360,54 @@ public HttpClientOptions setConnectionIdleTimeout(Duration connectionIdleTimeout
     /**
      * Gets the duration of time before an idle connection is closed.
      * <p>
-     * By default the connection idle timeout is 60 seconds.
+     * The default connection idle timeout is 60 seconds.
      *
      * @return The connection idle timeout duration.
      */
     public Duration getConnectionIdleTimeout() {
         return getTimeout(connectionIdleTimeout, DEFAULT_CONNECTION_IDLE_TIMEOUT);
     }
 
+    /**
+     * Sets the name of the {@link HttpClientProvider} implementation that should be used to construct an instance of
+     * {@link HttpClient}.
+     * <p>
+     * The name must be the full class name, ex {@code com.azure.core.http.netty.NettyAsyncHttpClientProvider} and not
+     * {@code NettyAsyncHttpClientProvider}, to disambiguate multiple providers with the same name but from different
+     * packages.
+     * <p>
+     * If the value isn't set or is an empty string the first {@link HttpClientProvider} found on the class path will
+     * be used to create an instance of {@link HttpClient}. If the value is set and doesn't match any
+     * {@link HttpClientProvider} found on the class path an {@link IllegalStateException} will be thrown when
+     * attempting to create an instance of {@link HttpClient}.
+     *
+     * @param httpClientImplementation The fully qualified name of an {@link HttpClientProvider} implementation.
+     * @return The updated HttpClientOptions object.
+     */
+    public HttpClientOptions setHttpClientImplementation(String httpClientImplementation) {
```

Comment:
What do you think about something like this:

```suggestion
    public <T extends HttpClientProvider> HttpClientOptions setHttpClientProvider(Class<T> provider)     
```

It might be more straightforward to use and less error prone.

As a side effect, you might even load the class without SPI, i.e. open the door to add test providers without dealing with module-info

Reply:
This could be done and then if the environment variable is configured to something that isn't found it'll throw during the creation of the class